applications/DeepSpeed-Chat/training/utils/perf.py

```diff
@@ -0,0 +1,61 @@
+# Copyright (c) Microsoft Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+# DeepSpeed Team
+
+import torch
+
+
+def print_throughput(hf_model,
+                     args,
+                     e2e_time,
+                     gen_exp_time,
+                     gen_seq_time,
+                     train_time,
+                     rank=0):
+    if rank <= 0:
+        hf_config = hf_model.config
+        num_layers = getattr(hf_config, "num_hidden_layers",
+                             getattr(hf_config, "n_layer", None))
+        hidden_size = getattr(hf_config, "hidden_size",
+                              getattr(hf_config, "n_embd", None))
+        vocab_size = getattr(hf_config, "vocab_size", None)
+        assert all(
+            (num_layers, hidden_size, vocab_size)
+        ), "Could not determine number of layers, hidden size, and vocab size of the model"
+
+        gpus_per_model = torch.distributed.get_world_size()
+        seq_length = args.max_answer_seq_len
+        batch_size = args.per_device_training_batch_size * gpus_per_model
```

Comment:
bs = args.per_device_generatation_batch_size * * args.generate_batch * args.ppo_epoch * gpus_per_model

Reply:
Gotcha. I will update this calculation.

Reply:
Offline discussion done and finalized the code.